### ISSUE TITLE
Invoking type() twice causes a failure of codon

### ISSUE THREAD

**Issue by xiaxinmeng** [exaloop/codon#243]
See the following example, we print the return value of function type() twice, codon fails. This case can work well on CPython. 

test.py
```
from random import randint
num = randint(1,10)
print(type(num))
guess = 0
print(type(guess)) 
```

Reproduce:
codon/codon-linux-x86_64/codon-deploy/bin/codon' run -release test.py

Crash Message:
```
Assert failed:	cannot find 'int.__new__:dispatch' [85608_1093878759__0.py:14:1]
Expression:	val
Source:		/github/workspace/codon/parser/visitors/typecheck/access.cpp:40
Aborted (core dumped)
```

Behavior on CPython 3.10.8:
```
<class 'int'>
<class 'int'>
```

Environment:
codon: v0.15.5 on Feb 6
Ubuntu 18.04

**Comment by gagson** [comment]
Got the exact same error written on https://github.com/exaloop/codon/issues/109#issuecomment-1482190119.
But I have nothing to do with type() or CPython. Wondering what is it.

**Comment by inumanag** [comment]
A bug on our end. Thank you for reporting it—will be handled in the next release.

**Comment by inumanag** [comment]
Fixed in #604.
